[C,PIL/PcxImagePlugin.py,147] 
[C,PIL/PcxImagePlugin.py,148]     # under windows, we could determine the current screen size with
[C,PIL/PcxImagePlugin.py,149]     # "Image.core.display_mode()[1]", but I think that's overkill...
[C,PIL/PcxImagePlugin.py,150] 
[C,PIL/PcxImagePlugin.py,151]     screen = im.size
[C,PIL/PcxImagePlugin.py,152] 
[C,PIL/PcxImagePlugin.py,153]     dpi = 100, 100
@@ -92,32 +92,32 @@ def _save(im, fp, filename):
[C,PIL/PdfImagePlugin.py,92]         procset = "/ImageB" # grayscale
[C,PIL/PdfImagePlugin.py,93]         bits = 1
[C,PIL/PdfImagePlugin.py,94]     elif im.mode == "L":
[C,PIL/PdfImagePlugin.py,95]         filter = "/DCTDecode"
[C,PIL/PdfImagePlugin.py,96]         # params = "<< /Predictor 15 /Columns %d >>" % (width-2)
[C,PIL/PdfImagePlugin.py,97]         colorspace = "/DeviceGray"
[C,PIL/PdfImagePlugin.py,98]         procset = "/ImageB" # grayscale
[C,PIL/PdfImagePlugin.py,99]     elif im.mode == "P":
[C,PIL/PdfImagePlugin.py,100]         filter = "/ASCIIHexDecode"
[C,PIL/PdfImagePlugin.py,101]         colorspace = "[ /Indexed /DeviceRGB 255 <"
[C,PIL/PdfImagePlugin.py,102]         palette = im.im.getpalette("RGB")
[C,PIL/PdfImagePlugin.py,103]         for i in range(256):
[C,PIL/PdfImagePlugin.py,104]             r = i8(palette[i*3])
[C,PIL/PdfImagePlugin.py,105]             g = i8(palette[i*3+1])
[C,PIL/PdfImagePlugin.py,106]             b = i8(palette[i*3+2])
[D,PIL/PdfImagePlugin.py,107]             colorspace = colorspace + "%02x%02x%02x " % (r, g, b)
[D,PIL/PdfImagePlugin.py,108]         colorspace = colorspace + b"> ]"
[A,PIL/PdfImagePlugin.py,107]             colorspace += "%02x%02x%02x " % (r, g, b)
[A,PIL/PdfImagePlugin.py,108]         colorspace += b"> ]"
[C,PIL/PdfImagePlugin.py,109]         procset = "/ImageI" # indexed color
[C,PIL/PdfImagePlugin.py,110]     elif im.mode == "RGB":
[C,PIL/PdfImagePlugin.py,111]         filter = "/DCTDecode"
[C,PIL/PdfImagePlugin.py,112]         colorspace = "/DeviceRGB"
[C,PIL/PdfImagePlugin.py,113]         procset = "/ImageC" # color images
[C,PIL/PdfImagePlugin.py,114]     elif im.mode == "CMYK":
[C,PIL/PdfImagePlugin.py,115]         filter = "/DCTDecode"
[C,PIL/PdfImagePlugin.py,116]         colorspace = "/DeviceCMYK"
[C,PIL/PdfImagePlugin.py,117]         procset = "/ImageC" # color images
[C,PIL/PdfImagePlugin.py,118]     else:
[C,PIL/PdfImagePlugin.py,119]         raise ValueError("cannot save mode %s" % im.mode)
[C,PIL/PdfImagePlugin.py,120] 
[C,PIL/PdfImagePlugin.py,121]     #
[C,PIL/PdfImagePlugin.py,122]     # catalogue
[C,PIL/PdfImagePlugin.py,123] 
@@ -223,73 +223,73 @@ def _layerinfo(file):
[C,PIL/PsdImagePlugin.py,223]                 name = read(length).decode('latin-1', 'replace')
[C,PIL/PsdImagePlugin.py,224]             combined += length + 1
[C,PIL/PsdImagePlugin.py,225] 
[C,PIL/PsdImagePlugin.py,226]         file.seek(size - combined, 1)
[C,PIL/PsdImagePlugin.py,227]         layers.append((name, mode, (x0, y0, x1, y1)))
[C,PIL/PsdImagePlugin.py,228] 
[C,PIL/PsdImagePlugin.py,229]     # get tiles
[C,PIL/PsdImagePlugin.py,230]     i = 0
[C,PIL/PsdImagePlugin.py,231]     for name, mode, bbox in layers:
[C,PIL/PsdImagePlugin.py,232]         tile = []
[C,PIL/PsdImagePlugin.py,233]         for m in mode:
[C,PIL/PsdImagePlugin.py,234]             t = _maketile(file, m, bbox, 1)
[C,PIL/PsdImagePlugin.py,235]             if t:
[C,PIL/PsdImagePlugin.py,236]                 tile.extend(t)
[C,PIL/PsdImagePlugin.py,237]         layers[i] = name, mode, bbox, tile
[D,PIL/PsdImagePlugin.py,238]         i = i + 1
[A,PIL/PsdImagePlugin.py,238]         i += 1
[C,PIL/PsdImagePlugin.py,239] 
[C,PIL/PsdImagePlugin.py,240]     return layers
[C,PIL/PsdImagePlugin.py,241] 
[C,PIL/PsdImagePlugin.py,242] def _maketile(file, mode, bbox, channels):
[C,PIL/PsdImagePlugin.py,243] 
[C,PIL/PsdImagePlugin.py,244]     tile = None
[C,PIL/PsdImagePlugin.py,245]     read = file.read
[C,PIL/PsdImagePlugin.py,246] 
[C,PIL/PsdImagePlugin.py,247]     compression = i16(read(2))
[C,PIL/PsdImagePlugin.py,248] 
[C,PIL/PsdImagePlugin.py,249]     xsize = bbox[2] - bbox[0]
[C,PIL/PsdImagePlugin.py,250]     ysize = bbox[3] - bbox[1]
[C,PIL/PsdImagePlugin.py,251] 
[C,PIL/PsdImagePlugin.py,252]     offset = file.tell()
[C,PIL/PsdImagePlugin.py,253] 
[C,PIL/PsdImagePlugin.py,254]     if compression == 0:
[C,PIL/PsdImagePlugin.py,255]         #
[C,PIL/PsdImagePlugin.py,256]         # raw compression
[C,PIL/PsdImagePlugin.py,257]         tile = []
[C,PIL/PsdImagePlugin.py,258]         for channel in range(channels):
[C,PIL/PsdImagePlugin.py,259]             layer = mode[channel]
[C,PIL/PsdImagePlugin.py,260]             if mode == "CMYK":
[D,PIL/PsdImagePlugin.py,261]                 layer = layer + ";I"
[A,PIL/PsdImagePlugin.py,261]                 layer += ";I"
[C,PIL/PsdImagePlugin.py,262]             tile.append(("raw", bbox, offset, layer))
[C,PIL/PsdImagePlugin.py,263]             offset = offset + xsize*ysize
[C,PIL/PsdImagePlugin.py,264] 
[C,PIL/PsdImagePlugin.py,265]     elif compression == 1:
[C,PIL/PsdImagePlugin.py,266]         #
[C,PIL/PsdImagePlugin.py,267]         # packbits compression
[C,PIL/PsdImagePlugin.py,268]         i = 0
[C,PIL/PsdImagePlugin.py,269]         tile = []
[C,PIL/PsdImagePlugin.py,270]         bytecount = read(channels * ysize * 2)
[C,PIL/PsdImagePlugin.py,271]         offset = file.tell()
[C,PIL/PsdImagePlugin.py,272]         for channel in range(channels):
[C,PIL/PsdImagePlugin.py,273]             layer = mode[channel]
[C,PIL/PsdImagePlugin.py,274]             if mode == "CMYK":
[D,PIL/PsdImagePlugin.py,275]                 layer = layer + ";I"
[A,PIL/PsdImagePlugin.py,275]                 layer += ";I"
[C,PIL/PsdImagePlugin.py,276]             tile.append(
[C,PIL/PsdImagePlugin.py,277]                 ("packbits", bbox, offset, layer)
[C,PIL/PsdImagePlugin.py,278]                 )
[C,PIL/PsdImagePlugin.py,279]             for y in range(ysize):
[C,PIL/PsdImagePlugin.py,280]                 offset = offset + i16(bytecount[i:i+2])
[D,PIL/PsdImagePlugin.py,281]                 i = i + 2
[A,PIL/PsdImagePlugin.py,281]                 i += 2
[C,PIL/PsdImagePlugin.py,282] 
[C,PIL/PsdImagePlugin.py,283]     file.seek(offset)
[C,PIL/PsdImagePlugin.py,284] 
[C,PIL/PsdImagePlugin.py,285]     if offset & 1:
[C,PIL/PsdImagePlugin.py,286]         read(1) # padding
[C,PIL/PsdImagePlugin.py,287] 
[C,PIL/PsdImagePlugin.py,288]     return tile
[C,PIL/PsdImagePlugin.py,289] 
[C,PIL/PsdImagePlugin.py,290] # --------------------------------------------------------------------
[C,PIL/PsdImagePlugin.py,291] # registry
[C,PIL/PsdImagePlugin.py,292] 
[C,PIL/PsdImagePlugin.py,293] Image.register_open("PSD", PsdImageFile, _accept)
[C,PIL/PsdImagePlugin.py,294] 
[C,PIL/PsdImagePlugin.py,295] Image.register_extension("PSD", ".psd")
@@ -546,33 +546,33 @@ class ImageFileDirectory(collections.MutableMapping):
[C,PIL/TiffImagePlugin.py,546]                     print("- value: <table: %d bytes>" % size)
[C,PIL/TiffImagePlugin.py,547]                 else:
[C,PIL/TiffImagePlugin.py,548]                     print("- value:", value)
[C,PIL/TiffImagePlugin.py,549] 
[C,PIL/TiffImagePlugin.py,550]             # figure out if data fits into the directory
[C,PIL/TiffImagePlugin.py,551]             if len(data) == 4:
[C,PIL/TiffImagePlugin.py,552]                 append((tag, typ, len(value), data, b""))
[C,PIL/TiffImagePlugin.py,553]             elif len(data) < 4:
[C,PIL/TiffImagePlugin.py,554]                 append((tag, typ, len(value), data + (4-len(data))*b"\0", b""))
[C,PIL/TiffImagePlugin.py,555]             else:
[C,PIL/TiffImagePlugin.py,556]                 count = len(value)
[C,PIL/TiffImagePlugin.py,557]                 if typ == 5:
[C,PIL/TiffImagePlugin.py,558]                     count = count // 2        # adjust for rational data field
[C,PIL/TiffImagePlugin.py,559] 
[C,PIL/TiffImagePlugin.py,560]                 append((tag, typ, count, o32(offset), data))
[D,PIL/TiffImagePlugin.py,561]                 offset = offset + len(data)
[A,PIL/TiffImagePlugin.py,561]                 offset += len(data)
[C,PIL/TiffImagePlugin.py,562]                 if offset & 1:
[D,PIL/TiffImagePlugin.py,563]                     offset = offset + 1 # word padding
[A,PIL/TiffImagePlugin.py,563]                     offset += 1  # word padding
[C,PIL/TiffImagePlugin.py,564] 
[C,PIL/TiffImagePlugin.py,565]         # update strip offset data to point beyond auxiliary data
[C,PIL/TiffImagePlugin.py,566]         if stripoffsets is not None:
[C,PIL/TiffImagePlugin.py,567]             tag, typ, count, value, data = directory[stripoffsets]
[C,PIL/TiffImagePlugin.py,568]             assert not data, "multistrip support not yet implemented"
[C,PIL/TiffImagePlugin.py,569]             value = o32(self.i32(value) + offset)
[C,PIL/TiffImagePlugin.py,570]             directory[stripoffsets] = tag, typ, count, value, data
[C,PIL/TiffImagePlugin.py,571] 
[C,PIL/TiffImagePlugin.py,572]         # pass 2: write directory to file
[C,PIL/TiffImagePlugin.py,573]         for tag, typ, count, value, data in directory:
[C,PIL/TiffImagePlugin.py,574]             if Image.DEBUG > 1:
[C,PIL/TiffImagePlugin.py,575]                 print(tag, typ, count, repr(value), repr(data))
[C,PIL/TiffImagePlugin.py,576]             fp.write(o16(tag) + o16(typ) + o32(count) + value)
[C,PIL/TiffImagePlugin.py,577] 
[C,PIL/TiffImagePlugin.py,578]         # -- overwrite here for multi-page --
@@ -632,31 +632,31 @@ class TiffImageFile(ImageFile.ImageFile):
[C,PIL/TiffImagePlugin.py,632]         return self._tell()
[C,PIL/TiffImagePlugin.py,633] 
[C,PIL/TiffImagePlugin.py,634]     def _seek(self, frame):
[C,PIL/TiffImagePlugin.py,635] 
[C,PIL/TiffImagePlugin.py,636]         self.fp = self.__fp
[C,PIL/TiffImagePlugin.py,637]         if frame < self.__frame:
[C,PIL/TiffImagePlugin.py,638]             # rewind file
[C,PIL/TiffImagePlugin.py,639]             self.__frame = -1
[C,PIL/TiffImagePlugin.py,640]             self.__next = self.__first
[C,PIL/TiffImagePlugin.py,641]         while self.__frame < frame:
[C,PIL/TiffImagePlugin.py,642]             if not self.__next:
[C,PIL/TiffImagePlugin.py,643]                 raise EOFError("no more images in TIFF file")
[C,PIL/TiffImagePlugin.py,644]             self.fp.seek(self.__next)
[C,PIL/TiffImagePlugin.py,645]             self.tag.load(self.fp)
[C,PIL/TiffImagePlugin.py,646]             self.__next = self.tag.next
[D,PIL/TiffImagePlugin.py,647]             self.__frame = self.__frame + 1
[A,PIL/TiffImagePlugin.py,647]             self.__frame += 1
[C,PIL/TiffImagePlugin.py,648]         self._setup()
[C,PIL/TiffImagePlugin.py,649] 
[C,PIL/TiffImagePlugin.py,650]     def _tell(self):
[C,PIL/TiffImagePlugin.py,651] 
[C,PIL/TiffImagePlugin.py,652]         return self.__frame
[C,PIL/TiffImagePlugin.py,653] 
[C,PIL/TiffImagePlugin.py,654]     def _decoder(self, rawmode, layer, tile=None):
[C,PIL/TiffImagePlugin.py,655]         "Setup decoder contexts"
[C,PIL/TiffImagePlugin.py,656] 
[C,PIL/TiffImagePlugin.py,657]         args = None
[C,PIL/TiffImagePlugin.py,658]         if rawmode == "RGB" and self._planar_configuration == 2:
[C,PIL/TiffImagePlugin.py,659]             rawmode = rawmode[layer]
[C,PIL/TiffImagePlugin.py,660]         compression = self._compression
[C,PIL/TiffImagePlugin.py,661]         if compression == "raw":
[C,PIL/TiffImagePlugin.py,662]             args = (rawmode, 0, 1)
@@ -888,52 +888,52 @@ class TiffImageFile(ImageFile.ImageFile):
[C,PIL/TiffImagePlugin.py,888]                      0, a))
[C,PIL/TiffImagePlugin.py,889]                 a = None
[C,PIL/TiffImagePlugin.py,890] 
[C,PIL/TiffImagePlugin.py,891]             else:
[C,PIL/TiffImagePlugin.py,892]                 for i in range(len(offsets)):
[C,PIL/TiffImagePlugin.py,893]                     a = self._decoder(rawmode, l, i)
[C,PIL/TiffImagePlugin.py,894]                     self.tile.append(
[C,PIL/TiffImagePlugin.py,895]                         (self._compression,
[C,PIL/TiffImagePlugin.py,896]                         (0, min(y, ysize), w, min(y+h, ysize)),
[C,PIL/TiffImagePlugin.py,897]                         offsets[i], a))
[C,PIL/TiffImagePlugin.py,898]                     if Image.DEBUG:
[C,PIL/TiffImagePlugin.py,899]                         print ("tiles: ", self.tile)
[C,PIL/TiffImagePlugin.py,900]                     y = y + h
[C,PIL/TiffImagePlugin.py,901]                     if y >= self.size[1]:
[C,PIL/TiffImagePlugin.py,902]                         x = y = 0
[D,PIL/TiffImagePlugin.py,903]                         l = l + 1
[A,PIL/TiffImagePlugin.py,903]                         l += 1
[C,PIL/TiffImagePlugin.py,904]                     a = None
[C,PIL/TiffImagePlugin.py,905]         elif TILEOFFSETS in self.tag:
[C,PIL/TiffImagePlugin.py,906]             # tiled image
[C,PIL/TiffImagePlugin.py,907]             w = getscalar(322)
[C,PIL/TiffImagePlugin.py,908]             h = getscalar(323)
[C,PIL/TiffImagePlugin.py,909]             a = None
[C,PIL/TiffImagePlugin.py,910]             for o in self.tag[TILEOFFSETS]:
[C,PIL/TiffImagePlugin.py,911]                 if not a:
[C,PIL/TiffImagePlugin.py,912]                     a = self._decoder(rawmode, l)
[C,PIL/TiffImagePlugin.py,913]                 # FIXME: this doesn't work if the image size
[C,PIL/TiffImagePlugin.py,914]                 # is not a multiple of the tile size...
[C,PIL/TiffImagePlugin.py,915]                 self.tile.append(
[C,PIL/TiffImagePlugin.py,916]                     (self._compression,
[C,PIL/TiffImagePlugin.py,917]                     (x, y, x+w, y+h),
[C,PIL/TiffImagePlugin.py,918]                     o, a))
[C,PIL/TiffImagePlugin.py,919]                 x = x + w
[C,PIL/TiffImagePlugin.py,920]                 if x >= self.size[0]:
[C,PIL/TiffImagePlugin.py,921]                     x, y = 0, y + h
[C,PIL/TiffImagePlugin.py,922]                     if y >= self.size[1]:
[C,PIL/TiffImagePlugin.py,923]                         x = y = 0
[D,PIL/TiffImagePlugin.py,924]                         l = l + 1
[A,PIL/TiffImagePlugin.py,924]                         l += 1
[C,PIL/TiffImagePlugin.py,925]                         a = None
[C,PIL/TiffImagePlugin.py,926]         else:
[C,PIL/TiffImagePlugin.py,927]             if Image.DEBUG:
[C,PIL/TiffImagePlugin.py,928]                 print("- unsupported data organization")
[C,PIL/TiffImagePlugin.py,929]             raise SyntaxError("unknown data organization")
[C,PIL/TiffImagePlugin.py,930] 
[C,PIL/TiffImagePlugin.py,931]         # fixup palette descriptor
[C,PIL/TiffImagePlugin.py,932] 
[C,PIL/TiffImagePlugin.py,933]         if self.mode == "P":
[C,PIL/TiffImagePlugin.py,934]             palette = [o8(a // 256) for a in self.tag[COLORMAP]]
[C,PIL/TiffImagePlugin.py,935]             self.palette = ImagePalette.raw("RGB;L", b"".join(palette))
[C,PIL/TiffImagePlugin.py,936] #
[C,PIL/TiffImagePlugin.py,937] # --------------------------------------------------------------------
[C,PIL/TiffImagePlugin.py,938] # Write TIFF files
[C,PIL/TiffImagePlugin.py,939] 
@@ -1081,31 +1081,31 @@ def _save(im, fp, filename):
[C,PIL/TiffImagePlugin.py,1081]     stride = len(bits) * ((im.size[0]*bits[0]+7)//8)
[C,PIL/TiffImagePlugin.py,1082]     ifd[ROWSPERSTRIP] = im.size[1]
[C,PIL/TiffImagePlugin.py,1083]     ifd[STRIPBYTECOUNTS] = stride * im.size[1]
[C,PIL/TiffImagePlugin.py,1084]     ifd[STRIPOFFSETS] = 0 # this is adjusted by IFD writer
[C,PIL/TiffImagePlugin.py,1085]     ifd[COMPRESSION] = COMPRESSION_INFO_REV.get(compression,1) # no compression by default
[C,PIL/TiffImagePlugin.py,1086] 
[C,PIL/TiffImagePlugin.py,1087]     if libtiff:
[C,PIL/TiffImagePlugin.py,1088]         if Image.DEBUG:
[C,PIL/TiffImagePlugin.py,1089]             print ("Saving using libtiff encoder")
[C,PIL/TiffImagePlugin.py,1090]             print (ifd.items())
[C,PIL/TiffImagePlugin.py,1091]         _fp = 0
[C,PIL/TiffImagePlugin.py,1092]         if hasattr(fp, "fileno"):
[C,PIL/TiffImagePlugin.py,1093]             fp.seek(0)
[C,PIL/TiffImagePlugin.py,1094]             _fp = os.dup(fp.fileno())
[C,PIL/TiffImagePlugin.py,1095] 
[D,PIL/TiffImagePlugin.py,1096]         blocklist =  [STRIPOFFSETS, STRIPBYTECOUNTS, ROWSPERSTRIP, ICCPROFILE] # ICC Profile crashes.
[A,PIL/TiffImagePlugin.py,1096]         blocklist = [STRIPOFFSETS, STRIPBYTECOUNTS, ROWSPERSTRIP, ICCPROFILE] # ICC Profile crashes.
[C,PIL/TiffImagePlugin.py,1097]         atts={}
[C,PIL/TiffImagePlugin.py,1098]         # bits per sample is a single short in the tiff directory, not a list. 
[C,PIL/TiffImagePlugin.py,1099]         atts[BITSPERSAMPLE] = bits[0]
[C,PIL/TiffImagePlugin.py,1100]         # Merge the ones that we have with (optional) more bits from
[C,PIL/TiffImagePlugin.py,1101]         # the original file, e.g x,y resolution so that we can
[C,PIL/TiffImagePlugin.py,1102]         # save(load('')) == original file.
[C,PIL/TiffImagePlugin.py,1103]         for k,v in itertools.chain(ifd.items(), getattr(im, 'ifd', {}).items()):
[C,PIL/TiffImagePlugin.py,1104]             if k not in atts and k not in blocklist:
[C,PIL/TiffImagePlugin.py,1105]                 if type(v[0]) == tuple and len(v) > 1:
[C,PIL/TiffImagePlugin.py,1106]                     # A tuple of more than one rational tuples
[C,PIL/TiffImagePlugin.py,1107]                     # flatten to floats, following tiffcp.c->cpTag->TIFF_RATIONAL
[C,PIL/TiffImagePlugin.py,1108]                     atts[k] = [float(elt[0])/float(elt[1]) for elt in v]
[C,PIL/TiffImagePlugin.py,1109]                     continue
[C,PIL/TiffImagePlugin.py,1110]                 if type(v[0]) == tuple and len(v) == 1:
[C,PIL/TiffImagePlugin.py,1111]                     # A tuple of one rational tuples
@@ -149,55 +149,55 @@ TAGS = {
[C,PIL/TiffTags.py,149] 
[C,PIL/TiffTags.py,150]     # Adobe DNG
[C,PIL/TiffTags.py,151]     50706: "DNGVersion",
[C,PIL/TiffTags.py,152]     50707: "DNGBackwardVersion",
[C,PIL/TiffTags.py,153]     50708: "UniqueCameraModel",
[C,PIL/TiffTags.py,154]     50709: "LocalizedCameraModel",
[C,PIL/TiffTags.py,155]     50710: "CFAPlaneColor",
[C,PIL/TiffTags.py,156]     50711: "CFALayout",
[C,PIL/TiffTags.py,157]     50712: "LinearizationTable",
[C,PIL/TiffTags.py,158]     50713: "BlackLevelRepeatDim",
[C,PIL/TiffTags.py,159]     50714: "BlackLevel",
[C,PIL/TiffTags.py,160]     50715: "BlackLevelDeltaH",
[C,PIL/TiffTags.py,161]     50716: "BlackLevelDeltaV",
[C,PIL/TiffTags.py,162]     50717: "WhiteLevel",
[C,PIL/TiffTags.py,163]     50718: "DefaultScale",
[D,PIL/TiffTags.py,164]     50741: "BestQualityScale",
[A,PIL/TiffTags.py,164]     50741: "BestQualityScale",  # FIXME! Dictionary contains duplicate keys 50741
[C,PIL/TiffTags.py,165]     50719: "DefaultCropOrigin",
[C,PIL/TiffTags.py,166]     50720: "DefaultCropSize",
[C,PIL/TiffTags.py,167]     50778: "CalibrationIlluminant1",
[C,PIL/TiffTags.py,168]     50779: "CalibrationIlluminant2",
[C,PIL/TiffTags.py,169]     50721: "ColorMatrix1",
[C,PIL/TiffTags.py,170]     50722: "ColorMatrix2",
[C,PIL/TiffTags.py,171]     50723: "CameraCalibration1",
[C,PIL/TiffTags.py,172]     50724: "CameraCalibration2",
[C,PIL/TiffTags.py,173]     50725: "ReductionMatrix1",
[C,PIL/TiffTags.py,174]     50726: "ReductionMatrix2",
[C,PIL/TiffTags.py,175]     50727: "AnalogBalance",
[C,PIL/TiffTags.py,176]     50728: "AsShotNeutral",
[C,PIL/TiffTags.py,177]     50729: "AsShotWhiteXY",
[C,PIL/TiffTags.py,178]     50730: "BaselineExposure",
[C,PIL/TiffTags.py,179]     50731: "BaselineNoise",
[C,PIL/TiffTags.py,180]     50732: "BaselineSharpness",
[C,PIL/TiffTags.py,181]     50733: "BayerGreenSplit",
[C,PIL/TiffTags.py,182]     50734: "LinearResponseLimit",
[C,PIL/TiffTags.py,183]     50735: "CameraSerialNumber",
[C,PIL/TiffTags.py,184]     50736: "LensInfo",
[C,PIL/TiffTags.py,185]     50737: "ChromaBlurRadius",
[C,PIL/TiffTags.py,186]     50738: "AntiAliasStrength",
[C,PIL/TiffTags.py,187]     50740: "DNGPrivateData",
[D,PIL/TiffTags.py,188]     50741: "MakerNoteSafety",
[A,PIL/TiffTags.py,188]     50741: "MakerNoteSafety",  # FIXME! Dictionary contains duplicate keys 50741
[C,PIL/TiffTags.py,189] 
[C,PIL/TiffTags.py,190]     #ImageJ
[C,PIL/TiffTags.py,191]     50838: "ImageJMetaDataByteCounts", # private tag registered with Adobe
[C,PIL/TiffTags.py,192]     50839: "ImageJMetaData", # private tag registered with Adobe
[C,PIL/TiffTags.py,193] }
[C,PIL/TiffTags.py,194] 
[C,PIL/TiffTags.py,195] ##
[C,PIL/TiffTags.py,196] # Map type numbers to type names.
[C,PIL/TiffTags.py,197] 
[C,PIL/TiffTags.py,198] TYPES = {
[C,PIL/TiffTags.py,199] 
[C,PIL/TiffTags.py,200]     1: "byte",
[C,PIL/TiffTags.py,201]     2: "ascii",
[C,PIL/TiffTags.py,202]     3: "short",
[C,PIL/TiffTags.py,203]     4: "long",
@@ -47,31 +47,31 @@ if hasattr(Image.core, "drawwmf"):
[C,PIL/WmfImagePlugin.py,47]             return Image.frombytes(
[C,PIL/WmfImagePlugin.py,48]                 "RGB", im.size,
[C,PIL/WmfImagePlugin.py,49]                 Image.core.drawwmf(im.fp.read(), im.size, self.bbox),
[C,PIL/WmfImagePlugin.py,50]                 "raw", "BGR", (im.size[0]*3 + 3) & -4, -1
[C,PIL/WmfImagePlugin.py,51]                 )
[C,PIL/WmfImagePlugin.py,52] 
[C,PIL/WmfImagePlugin.py,53]     register_handler(WmfHandler())
[C,PIL/WmfImagePlugin.py,54] 
[C,PIL/WmfImagePlugin.py,55] # --------------------------------------------------------------------
[C,PIL/WmfImagePlugin.py,56] 
[C,PIL/WmfImagePlugin.py,57] word = _binary.i16le
[C,PIL/WmfImagePlugin.py,58] 
[C,PIL/WmfImagePlugin.py,59] def short(c, o=0):
[C,PIL/WmfImagePlugin.py,60]     v = word(c, o)
[C,PIL/WmfImagePlugin.py,61]     if v >= 32768:
[D,PIL/WmfImagePlugin.py,62]         v = v - 65536
[A,PIL/WmfImagePlugin.py,62]         v -= 65536
[C,PIL/WmfImagePlugin.py,63]     return v
[C,PIL/WmfImagePlugin.py,64] 
[C,PIL/WmfImagePlugin.py,65] dword = _binary.i32le
[C,PIL/WmfImagePlugin.py,66] 
[C,PIL/WmfImagePlugin.py,67] #
[C,PIL/WmfImagePlugin.py,68] # --------------------------------------------------------------------
[C,PIL/WmfImagePlugin.py,69] # Read WMF file
[C,PIL/WmfImagePlugin.py,70] 
[C,PIL/WmfImagePlugin.py,71] def _accept(prefix):
[C,PIL/WmfImagePlugin.py,72]     return (
[C,PIL/WmfImagePlugin.py,73]         prefix[:6] == b"\xd7\xcd\xc6\x9a\x00\x00" or
[C,PIL/WmfImagePlugin.py,74]         prefix[:4] == b"\x01\x00\x00\x00"
[C,PIL/WmfImagePlugin.py,75]         )
[C,PIL/WmfImagePlugin.py,76] 
[C,PIL/WmfImagePlugin.py,77] ##
@@ -92,19 +92,19 @@ if html:
[C,Scripts/explode.py,92] 
[C,Scripts/explode.py,93] while True:
[C,Scripts/explode.py,94] 
[C,Scripts/explode.py,95]     if frames[ix]:
[C,Scripts/explode.py,96]         im.save(outfile % ix)
[C,Scripts/explode.py,97]         print(outfile % ix)
[C,Scripts/explode.py,98] 
[C,Scripts/explode.py,99]         if html:
[C,Scripts/explode.py,100]             html.write("<img src='%s'><br>\n" % outfile % ix)
[C,Scripts/explode.py,101] 
[C,Scripts/explode.py,102]     try:
[C,Scripts/explode.py,103]         im.seek(ix)
[C,Scripts/explode.py,104]     except EOFError:
[C,Scripts/explode.py,105]         break
[C,Scripts/explode.py,106] 
[D,Scripts/explode.py,107]     ix = ix + 1
[A,Scripts/explode.py,107]     ix += 1
[C,Scripts/explode.py,108] 
[C,Scripts/explode.py,109] if html:
[C,Scripts/explode.py,110]     html.write("</body>\n</html>\n")
@@ -88,31 +88,31 @@ def makedelta(fp, sequence):
[C,Scripts/gifmaker.py,88] 
[C,Scripts/gifmaker.py,89]             bbox = delta.getbbox()
[C,Scripts/gifmaker.py,90] 
[C,Scripts/gifmaker.py,91]             if bbox:
[C,Scripts/gifmaker.py,92] 
[C,Scripts/gifmaker.py,93]                 # compress difference
[C,Scripts/gifmaker.py,94]                 for s in getdata(im.crop(bbox), offset = bbox[:2]):
[C,Scripts/gifmaker.py,95]                     fp.write(s)
[C,Scripts/gifmaker.py,96] 
[C,Scripts/gifmaker.py,97]             else:
[C,Scripts/gifmaker.py,98]                 # FIXME: what should we do in this case?
[C,Scripts/gifmaker.py,99]                 pass
[C,Scripts/gifmaker.py,100] 
[C,Scripts/gifmaker.py,101]         previous = im.copy()
[C,Scripts/gifmaker.py,102] 
[D,Scripts/gifmaker.py,103]         frames = frames + 1
[A,Scripts/gifmaker.py,103]         frames += 1
[C,Scripts/gifmaker.py,104] 
[C,Scripts/gifmaker.py,105]     fp.write(";")
[C,Scripts/gifmaker.py,106] 
[C,Scripts/gifmaker.py,107]     return frames
[C,Scripts/gifmaker.py,108] 
[C,Scripts/gifmaker.py,109] # --------------------------------------------------------------------
[C,Scripts/gifmaker.py,110] # main hack
[C,Scripts/gifmaker.py,111] 
[C,Scripts/gifmaker.py,112] def compress(infile, outfile):
[C,Scripts/gifmaker.py,113] 
[C,Scripts/gifmaker.py,114]     # open input image, and force loading of first frame
[C,Scripts/gifmaker.py,115]     im = Image.open(infile)
[C,Scripts/gifmaker.py,116]     im.load()
[C,Scripts/gifmaker.py,117] 
[C,Scripts/gifmaker.py,118]     # open output file
@@ -474,55 +474,55 @@ class PILDriver:
[C,Scripts/pildriver.py,474]         enhancer = ImageEnhance.Sharpness(image)
[C,Scripts/pildriver.py,475]         self.push(enhancer.enhance(factor))
[C,Scripts/pildriver.py,476] 
[C,Scripts/pildriver.py,477]     # The interpreter loop
[C,Scripts/pildriver.py,478] 
[C,Scripts/pildriver.py,479]     def execute(self, list):
[C,Scripts/pildriver.py,480]         "Interpret a list of PILDriver commands."
[C,Scripts/pildriver.py,481]         list.reverse()
[C,Scripts/pildriver.py,482]         while len(list) > 0:
[C,Scripts/pildriver.py,483]             self.push(list[0])
[C,Scripts/pildriver.py,484]             list = list[1:]
[C,Scripts/pildriver.py,485]             if self.verbose:
[C,Scripts/pildriver.py,486]                 print("Stack: " + repr(self.stack))
[C,Scripts/pildriver.py,487]             top = self.top()
[C,Scripts/pildriver.py,488]             if not isinstance(top, str):
[D,Scripts/pildriver.py,489]                 continue;
[A,Scripts/pildriver.py,489]                 continue
[C,Scripts/pildriver.py,490]             funcname = "do_" + top
[C,Scripts/pildriver.py,491]             if not hasattr(self, funcname):
[C,Scripts/pildriver.py,492]                 continue
[C,Scripts/pildriver.py,493]             else:
[C,Scripts/pildriver.py,494]                 self.do_pop()
[C,Scripts/pildriver.py,495]                 func = getattr(self, funcname)
[C,Scripts/pildriver.py,496]                 func()
[C,Scripts/pildriver.py,497] 
[C,Scripts/pildriver.py,498] if __name__ == '__main__':
[C,Scripts/pildriver.py,499]     import sys
[C,Scripts/pildriver.py,500]     try:
[C,Scripts/pildriver.py,501]         import readline
[C,Scripts/pildriver.py,502]     except ImportError:
[C,Scripts/pildriver.py,503]         pass # not available on all platforms
[C,Scripts/pildriver.py,504] 
[C,Scripts/pildriver.py,505]     # If we see command-line arguments, interpret them as a stack state
[C,Scripts/pildriver.py,506]     # and execute.  Otherwise go interactive.
[C,Scripts/pildriver.py,507] 
[C,Scripts/pildriver.py,508]     driver = PILDriver()
[C,Scripts/pildriver.py,509]     if len(sys.argv[1:]) > 0:
[C,Scripts/pildriver.py,510]         driver.execute(sys.argv[1:])
[C,Scripts/pildriver.py,511]     else:
[C,Scripts/pildriver.py,512]         print("PILDriver says hello.")
[C,Scripts/pildriver.py,513]         while True:
[C,Scripts/pildriver.py,514]             try:
[C,Scripts/pildriver.py,515]                 if sys.version_info[0] >= 3:
[D,Scripts/pildriver.py,516]                     line = input('pildriver> ');
[A,Scripts/pildriver.py,516]                     line = input('pildriver> ')
[C,Scripts/pildriver.py,517]                 else:
[D,Scripts/pildriver.py,518]                     line = raw_input('pildriver> ');
[A,Scripts/pildriver.py,518]                     line = raw_input('pildriver> ')
[C,Scripts/pildriver.py,519]             except EOFError:
[C,Scripts/pildriver.py,520]                 print("\nPILDriver says goodbye.")
[C,Scripts/pildriver.py,521]                 break
[C,Scripts/pildriver.py,522]             driver.execute(line.split())
[C,Scripts/pildriver.py,523]             print(driver.stack)
[C,Scripts/pildriver.py,524] 
[C,Scripts/pildriver.py,525] # The following sets edit modes for GNU EMACS
[C,Scripts/pildriver.py,526] # Local Variables:
[C,Scripts/pildriver.py,527] # mode:python
[C,Scripts/pildriver.py,528] # End:
@@ -45,31 +45,31 @@ verbose = quiet = verify = 0
[C,Scripts/pilfile.py,45] for o, a in opt:
[C,Scripts/pilfile.py,46]     if o == "-f":
[C,Scripts/pilfile.py,47]         Image.init()
[C,Scripts/pilfile.py,48]         id = sorted(Image.ID)
[C,Scripts/pilfile.py,49]         print("Supported formats:")
[C,Scripts/pilfile.py,50]         for i in id:
[C,Scripts/pilfile.py,51]             print(i, end=' ')
[C,Scripts/pilfile.py,52]         sys.exit(1)
[C,Scripts/pilfile.py,53]     elif o == "-i":
[C,Scripts/pilfile.py,54]         verbose = 1
[C,Scripts/pilfile.py,55]     elif o == "-q":
[C,Scripts/pilfile.py,56]         quiet = 1
[C,Scripts/pilfile.py,57]     elif o == "-v":
[C,Scripts/pilfile.py,58]         verify = 1
[C,Scripts/pilfile.py,59]     elif o == "-D":
[D,Scripts/pilfile.py,60]         Image.DEBUG = Image.DEBUG + 1
[A,Scripts/pilfile.py,60]         Image.DEBUG += 1
[C,Scripts/pilfile.py,61] 
[C,Scripts/pilfile.py,62] def globfix(files):
[C,Scripts/pilfile.py,63]     # expand wildcards where necessary
[C,Scripts/pilfile.py,64]     if sys.platform == "win32":
[C,Scripts/pilfile.py,65]         out = []
[C,Scripts/pilfile.py,66]         for file in files:
[C,Scripts/pilfile.py,67]             if glob.has_magic(file):
[C,Scripts/pilfile.py,68]                 out.extend(glob.glob(file))
[C,Scripts/pilfile.py,69]             else:
[C,Scripts/pilfile.py,70]                 out.append(file)
[C,Scripts/pilfile.py,71]         return out
[C,Scripts/pilfile.py,72]     return files
[C,Scripts/pilfile.py,73] 
[C,Scripts/pilfile.py,74] for file in globfix(args):
[C,Scripts/pilfile.py,75]     try:
@@ -60,31 +60,31 @@ for file in files:
[C,Tests/run.py,60] 
[C,Tests/run.py,61]     result_lines = result.splitlines()
[C,Tests/run.py,62]     if len(result_lines):
[C,Tests/run.py,63]         if result_lines[0] == "ignore_all_except_last_line":
[C,Tests/run.py,64]             result = result_lines[-1]
[C,Tests/run.py,65] 
[C,Tests/run.py,66]     # Extract any ignore patterns
[C,Tests/run.py,67]     ignore_pats = ignore_re.findall(result)
[C,Tests/run.py,68]     result = ignore_re.sub('', result)
[C,Tests/run.py,69] 
[C,Tests/run.py,70]     try:
[C,Tests/run.py,71]         def fix_re(p):
[C,Tests/run.py,72]             if not p.startswith('^'):
[C,Tests/run.py,73]                 p = '^' + p
[C,Tests/run.py,74]             if not p.endswith('$'):
[D,Tests/run.py,75]                 p = p + '$'
[A,Tests/run.py,75]                 p += '$'
[C,Tests/run.py,76]             return p
[C,Tests/run.py,77] 
[C,Tests/run.py,78]         ignore_res = [re.compile(fix_re(p), re.MULTILINE) for p in ignore_pats]
[C,Tests/run.py,79]     except:
[C,Tests/run.py,80]         print('(bad ignore patterns %r)' % ignore_pats)
[C,Tests/run.py,81]         ignore_res = []
[C,Tests/run.py,82] 
[C,Tests/run.py,83]     for r in ignore_res:
[C,Tests/run.py,84]         result = r.sub('', result)
[C,Tests/run.py,85] 
[C,Tests/run.py,86]     result = result.strip()
[C,Tests/run.py,87] 
[C,Tests/run.py,88]     if result == "ok":
[C,Tests/run.py,89]         result = None
[C,Tests/run.py,90]     elif result == "skip":
@@ -92,31 +92,31 @@ for file in files:
[C,Tests/run.py,92]         skipped.append(test)
[C,Tests/run.py,93]         continue
[C,Tests/run.py,94]     elif not result:
[C,Tests/run.py,95]         result = "(no output)"
[C,Tests/run.py,96]     status = out.close()
[C,Tests/run.py,97]     if status or result:
[C,Tests/run.py,98]         if status:
[C,Tests/run.py,99]             print("=== error", status)
[C,Tests/run.py,100]         if result:
[C,Tests/run.py,101]             if result[-3:] == "\nok":
[C,Tests/run.py,102]                 # if there's an ok at the end, it's not really ok
[C,Tests/run.py,103]                 result = result[:-3]
[C,Tests/run.py,104]             print(result)
[C,Tests/run.py,105]         failed.append(test)
[C,Tests/run.py,106]     else:
[D,Tests/run.py,107]         success = success + 1
[A,Tests/run.py,107]         success += 1
[C,Tests/run.py,108] 
[C,Tests/run.py,109] print("-"*68)
[C,Tests/run.py,110] 
[C,Tests/run.py,111] temp_root = os.path.join(tempfile.gettempdir(), 'pillow-tests')
[C,Tests/run.py,112] tempfiles = glob.glob(os.path.join(temp_root, "temp_*"))
[C,Tests/run.py,113] if tempfiles:
[C,Tests/run.py,114]     print("===", "remaining temporary files")
[C,Tests/run.py,115]     for file in tempfiles:
[C,Tests/run.py,116]         print(file)
[C,Tests/run.py,117]     print("-"*68)
[C,Tests/run.py,118] 
[C,Tests/run.py,119] 
[C,Tests/run.py,120] def tests(n):
[C,Tests/run.py,121]     if n == 1:
[C,Tests/run.py,122]         return "1 test"
@@ -49,32 +49,32 @@ def test_mesh():
[C,Tests/test_image_transform.py,49]     checker.paste(scaled, (0,0))
[C,Tests/test_image_transform.py,50]     checker.paste(scaled, (w//2,h//2))
[C,Tests/test_image_transform.py,51]         
[C,Tests/test_image_transform.py,52]     assert_image_equal(transformed, checker) 
[C,Tests/test_image_transform.py,53] 
[C,Tests/test_image_transform.py,54]     # now, check to see that the extra area is (0,0,0,0)
[C,Tests/test_image_transform.py,55]     blank = Image.new('RGBA', (w//2,h//2), (0,0,0,0))
[C,Tests/test_image_transform.py,56] 
[C,Tests/test_image_transform.py,57]     assert_image_equal(blank, transformed.crop((w//2,0,w,h//2)))
[C,Tests/test_image_transform.py,58]     assert_image_equal(blank, transformed.crop((0,h//2,w//2,h)))
[C,Tests/test_image_transform.py,59] 
[C,Tests/test_image_transform.py,60] def _test_alpha_premult(op):
[C,Tests/test_image_transform.py,61]      # create image with half white, half black, with the black half transparent.
[C,Tests/test_image_transform.py,62]     # do op, 
[C,Tests/test_image_transform.py,63]     # there should be no darkness in the white section.
[D,Tests/test_image_transform.py,64]     im = Image.new('RGBA', (10,10), (0,0,0,0));
[D,Tests/test_image_transform.py,65]     im2 = Image.new('RGBA', (5,10), (255,255,255,255));
[A,Tests/test_image_transform.py,64]     im = Image.new('RGBA', (10,10), (0,0,0,0))
[A,Tests/test_image_transform.py,65]     im2 = Image.new('RGBA', (5,10), (255,255,255,255))
[C,Tests/test_image_transform.py,66]     im.paste(im2, (0,0))
[C,Tests/test_image_transform.py,67]     
[C,Tests/test_image_transform.py,68]     im = op(im, (40,10))
[C,Tests/test_image_transform.py,69]     im_background = Image.new('RGB', (40,10), (255,255,255))
[C,Tests/test_image_transform.py,70]     im_background.paste(im, (0,0), im)
[C,Tests/test_image_transform.py,71]     
[C,Tests/test_image_transform.py,72]     hist = im_background.histogram()
[C,Tests/test_image_transform.py,73]     assert_equal(40*10, hist[-1])
[C,Tests/test_image_transform.py,74] 
[C,Tests/test_image_transform.py,75]     
[C,Tests/test_image_transform.py,76] def test_alpha_premult_resize():
[C,Tests/test_image_transform.py,77] 
[C,Tests/test_image_transform.py,78]     def op (im, sz):
[C,Tests/test_image_transform.py,79]         return im.resize(sz, Image.LINEAR)
[C,Tests/test_image_transform.py,80]     
@@ -4,19 +4,19 @@ from PIL import Image
[C,Tests/test_imagesequence.py,4] from PIL import ImageSequence
[C,Tests/test_imagesequence.py,5] 
[C,Tests/test_imagesequence.py,6] def test_sanity():
[C,Tests/test_imagesequence.py,7] 
[C,Tests/test_imagesequence.py,8]     file = tempfile("temp.im")
[C,Tests/test_imagesequence.py,9] 
[C,Tests/test_imagesequence.py,10]     im = lena("RGB")
[C,Tests/test_imagesequence.py,11]     im.save(file)
[C,Tests/test_imagesequence.py,12] 
[C,Tests/test_imagesequence.py,13]     seq = ImageSequence.Iterator(im)
[C,Tests/test_imagesequence.py,14] 
[C,Tests/test_imagesequence.py,15]     index = 0
[C,Tests/test_imagesequence.py,16]     for frame in seq:
[C,Tests/test_imagesequence.py,17]         assert_image_equal(im, frame)
[C,Tests/test_imagesequence.py,18]         assert_equal(im.tell(), index)
[D,Tests/test_imagesequence.py,19]         index = index + 1
[A,Tests/test_imagesequence.py,19]         index += 1
[C,Tests/test_imagesequence.py,20] 
[C,Tests/test_imagesequence.py,21]     assert_equal(index, 1)
[C,Tests/test_imagesequence.py,22] 